File: JS/webgl_objects/renderer.js
import { class_type } from "../.fable/fable-library.3.0.0/Reflection.js";
import { Grid2D_$ctor_Z166BE183 } from "./grid2d.js";
import { VectorObject2D_$ctor_Z166BE183 } from "./vectorobject2d.js";
import { LineObjects2D_$ctor_22E93213 } from "./lineobjects2d.js";
import { LinePathObject2D_$ctor_3F122D1B } from "./linepathobject2d.js";
import { PerlinNoise1DObject2D_$ctor_C1A1B4 } from "./perlinnoise1dobject2d.js";
import { PerlinNoise2DObject2D_$ctor_C1A1B4 } from "./perlinnoise2dobject2d.js";

export class GlRenderer {
    constructor() {
    }
}

export function GlRenderer$reflection() {
    return class_type("Wil.Webgl.GlRenderer", void 0, GlRenderer);
}

export function GlRenderer_$ctor() {
    return new GlRenderer();
}

export function GlRenderer__Grid2D_Z270D9612(_, scene, size, layer, name, linkTo, parallaxCam, parallaxDistance) {
    return Grid2D_$ctor_Z166BE183(scene.Canvas.Params, scene, size, layer, name, linkTo, parallaxCam, parallaxDistance);
}

export function GlRenderer__Vector2D_Z270D9612(_, scene, vector, layer, name, linkTo, parallaxCam, parallaxDistance) {
    return VectorObject2D_$ctor_Z166BE183(scene.Canvas.Params, scene, vector, layer, name, linkTo, parallaxCam, parallaxDistance);
}

export function GlRenderer__Line2D_515305A0(_, scene, p1, p2, layer, name, linkTo, parallaxCam, parallaxDistance) {
    return LineObjects2D_$ctor_22E93213(scene.Canvas.Params, scene, p1, p2, layer, name, linkTo, parallaxCam, parallaxDistance);
}

export function GlRenderer__LinePath2D_5BF31CC8(_, scene, points, layer, name, linkTo, parallaxCam, parallaxDistance) {
    return LinePathObject2D_$ctor_3F122D1B(scene.Canvas.Params, scene, points, layer, name, linkTo, parallaxCam, parallaxDistance);
}

export function GlRenderer__PerlinNoise1DObject2D_Z4FCA04F9(_, scene, size, seed, layer, name, linkTo, parallaxCam, parallaxDistance) {
    return PerlinNoise1DObject2D_$ctor_C1A1B4(scene.Canvas.Params, scene, size, seed, layer, name, linkTo, parallaxCam, parallaxDistance);
}

export function GlRenderer__PerlinNoise2DObject2D_Z4FCA04F9(_, scene, size, seed, layer, name, linkTo, parallaxCam, parallaxDistance) {
    return PerlinNoise2DObject2D_$ctor_C1A1B4(scene.Canvas.Params, scene, size, seed, layer, name, linkTo, parallaxCam, parallaxDistance);
}

File: src/webgl_objects/renderer.fs
namespace Wil.Webgl

open Wil.Webgl.Data

type GlRenderer() =
  member _.Grid2D(scene, size, ?layer, ?name, ?linkTo, ?parallaxCam, ?parallaxDistance) =
    Grid2D(scene.Canvas.Params, scene, size, ?layer = layer, ?name = name, ?linkTo = linkTo, ?parallaxCam = parallaxCam, ?parallaxDistance = parallaxDistance)

  member _.Vector2D(scene, vector, ?layer, ?name, ?linkTo, ?parallaxCam, ?parallaxDistance) =
    VectorObject2D(scene.Canvas.Params, scene, vector, ?layer = layer, ?name = name, ?linkTo = linkTo, ?parallaxCam = parallaxCam, ?parallaxDistance = parallaxDistance)

  member _.Line2D(scene, p1, p2, ?layer, ?name, ?linkTo, ?parallaxCam, ?parallaxDistance) =
    LineObjects2D(scene.Canvas.Params, scene, p1, p2, ?layer = layer, ?name = name, ?linkTo = linkTo, ?parallaxCam = parallaxCam, ?parallaxDistance = parallaxDistance)
  
  member _.LinePath2D(scene, points, ?layer, ?name, ?linkTo, ?parallaxCam, ?parallaxDistance) =
    LinePathObject2D(scene.Canvas.Params, scene, points, ?layer = layer, ?name = name, ?linkTo = linkTo, ?parallaxCam = parallaxCam, ?parallaxDistance = parallaxDistance)

  member _.PerlinNoise1DObject2D(scene, size, ?seed, ?layer, ?name, ?linkTo, ?parallaxCam, ?parallaxDistance) =
    PerlinNoise1DObject2D(scene.Canvas.Params, scene, size, ?seed = seed, ?layer = layer, ?name = name, ?linkTo = linkTo, ?parallaxCam = parallaxCam, ?parallaxDistance = parallaxDistance)

  member _.PerlinNoise2DObject2D(scene, size, ?seed, ?layer, ?name, ?linkTo, ?parallaxCam, ?parallaxDistance) =
    PerlinNoise2DObject2D(scene.Canvas.Params, scene, size, ?seed = seed, ?layer = layer, ?name = name, ?linkTo = linkTo, ?parallaxCam = parallaxCam, ?parallaxDistance = parallaxDistance)
